Xledger.Collections/Memory/Extensions.cs

```diff
@@ -0,0 +1,27 @@
+using System.Buffers;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Xledger.Collections.Memory;
+
+public static class Extensions {
+    /// <summary>
+    /// Returns an IMemoryOwner`1 with a Memory property of len. Takes ownership
+    /// of memoryOwner and returns the underlying Memory object via that memoryOwner's
+    /// Dispose.
+    /// </summary>
+    /// <remarks>This is in part why an instance of EmptyMemoryOwner isn't
+    /// returned when len = 0. The memory in memoryOwner must be returned.</remarks>
+    public static IMemoryOwner<T> Slice<T>(this IMemoryOwner<T> memoryOwner, int len) {
+        return new SizedMemoryOwner<T>(memoryOwner, len);
+    }
```

Comment:
We might also need the one that allows passing in the start index. Here is how they do it in span:

https://learn.microsoft.com/en-us/dotnet/api/system.span-1.slice?view=net-9.0